### DISCUSSION TITLE
Nav RTH on Explorer LR (No Compass)

### DISCUSSION THREAD

**Discussion by DivingFalconFPV** [iNavFlight/inav#6907] [Ideas]
Is there a way? Make a magnetometer option for a like betaflight type NAV RTH. GPS direction. 
Betaflight defaults compass north when first plugging in. Gyro tells drone it is turning,. Not great arrow doesnt always point right but gets me in general direction to get video or signal back. 

I tried FAKE in magnetometer to get NAV RTH in modes but compass warning. So FAIL
Going to try GPSMAG for magnetometer but I am sure that will be a FAIL too. 
So only way is to change the GPS unit?

**Comment by avsaase** [comment]
Yes, you need a compass to use any of the navigation features in INAV.

**Reply by breadoven** [reply]
I was thinking about this the other day. I crashed recently using RTH because the magnetometer had mysteriously lost its calibration (> 90 degrees). Shot off in a direction 90 degs off from where it should have headed, ending up hitting some trees. Looking at the DVR afterwards the heading error was obvious but I missed it because I wasn't looking for it. Made me wonder why there's no check for compass error when you have sufficient ground speed flying in non Nav modes to get a valid GPS heading.

Just added something yesterday in fact that displays compass vs GPS heading error on the OSD to see how it varies during flight. Not tried it yet so don't know how useful it might be. However, I noticed in imu.c that multirotors don't seem to use the GPS heading at all, totally dependent on the compass., which surprised me (unless I've missed something).

If GPS heading was used it could flag a mismatch between compass and GPS heading resulting in a warning with Nav modes being inhibited. With the exception of RTH perhaps which could still be available in a degraded form similar to Betaflight GPS Rescue which works (maybe) without a compass just GPS ... obviously on the basis that with sufficient ground speed you have a valid GPS heading to use.

**Reply by DivingFalconFPV** [reply]
If they added that they could add a magnetometer GPS compass option on list.  Just warn people it sucks. It won't come directly home or back to its home point,  just the area.
